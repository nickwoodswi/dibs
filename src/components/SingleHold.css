.single-hold-selector{
    display: flex;
    width: 100%;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    margin-top: 15px;
    border-radius: 5px;
    border: 1px dotted red;
    padding: 15px;
}
select{
    width: 120px;
    height: 50px;
    font-size: 18px;
    margin: 5px;
    border: none;
}
.date-selector{
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    width: 100%;
}