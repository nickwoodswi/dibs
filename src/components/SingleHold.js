import React, { Component } from 'react'
import CurrentAdd from './CurrentAdd'
import MonthDayCounts from './MonthDayCounts'
import './SingleHold.css'

class SingleHold extends Component {
    static defaultProps = {
        appState: {}
    }
    render() {
        const months = Object.keys(MonthDayCounts)
        const days = MonthDayCounts[this.props.appState.holdmonth_start]
        return(
            <div className="single-hold-selector">
                FROM:
                <div className="date-selector">
                    <select key="start-month-selector" className="month-selector" onChange={e => this.props.handleSingleHold('holdmonth_start', 'holdmonth_end', e.target.value)}>
                        {months.map((month, idx) => {
                            return(
                                <option key={idx}>{month}</option>
                            )
                        })}
                    </select>
                    <select key="start-day-selector" className="day-selector" onChange={e => this.props.handleSingleHold('holdday_start', 'holdday_end', e.target.value)}>
                        <>{days}</>
                    </select>
                    <select key="start-year-selector" className="year-selector" onChange={e => this.props.handleSingleHold('holdyear_start', 'holdyear_end', e.target.value)}>
                        <option>2020</option>
                        <option>2021</option>
                    </select>
                </div>
                <CurrentAdd appState={this.props.appState} /><br/>
            </div>
        )
    }
}

export default SingleHold