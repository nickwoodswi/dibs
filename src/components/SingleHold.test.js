import React from 'react';
import { render } from '@testing-library/react';
import SingleHold from './SingleHold';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders single hold component', () => {
    render(<MemoryRouter><SingleHold /></MemoryRouter>);
});

it('SingleHold renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><SingleHold /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});