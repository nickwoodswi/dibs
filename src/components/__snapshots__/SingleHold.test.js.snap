// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleHold renders the UI as expected 1`] = `
<div
  className="single-hold-selector"
>
  FROM:
  <div
    className="date-selector"
  >
    <select
      className="month-selector"
      onChange={[Function]}
    >
      <option>
        January
      </option>
      <option>
        February
      </option>
      <option>
        March
      </option>
      <option>
        April
      </option>
      <option>
        May
      </option>
      <option>
        June
      </option>
      <option>
        July
      </option>
      <option>
        August
      </option>
      <option>
        September
      </option>
      <option>
        October
      </option>
      <option>
        November
      </option>
      <option>
        December
      </option>
    </select>
    <select
      className="day-selector"
      onChange={[Function]}
    />
    <select
      className="year-selector"
      onChange={[Function]}
    >
      <option>
        2020
      </option>
      <option>
        2021
      </option>
    </select>
  </div>
  <div
    className="current-add"
  >
    <div
      className="current-add-header"
    >
      CURRENTLY ADDING HOLDS FROM:
    </div>
     
    , 
     thru 
     
    , 
  </div>
  <br />
</div>
`;
